12-add-tupletools.md

```diff
@@ -0,0 +1,117 @@
+>
+>  - `TupleToolKinematic`, which fills the kinematic information of the decay.
+>  - `TupleToolPid`, which stores DLL and PID information of the particle.
+>  - `TupleToolANNPID`, which stores the new NeuralNet-based PID information of the particle.
+>  - `TupleToolGeometry`, which stores the geometrical variables (IP, vertex position, etc) of the particle.
+>  - `TupleToolEventInfo`, which stores general information (event number, run number, GPS time, etc) of the event.
+
+In order to add `TupleTools` to `dtt`, we have to use the `addTupleTool` method of `DecayTreeTuple` (only available when we have `from DecayTreeTuple.Configuration import *` in our script).
+This method instantiates the tool, adds it to the list of tools to execute and returns it.
+For example, if we want to fill the tracking information of our particles, we can add the `TupleToolTrackInfo` tool in the following way:
+
+```python
+track_tool = dtt.addTupleTool('TupleToolTrackInfo')
+```
+
+Some tools can be configured. For example, if we wanted further information from the tracks, such as the number of degrees of freedom of the track fit, we would have to turn on the verbose mode of the tool:
+
+```python
+track_tool.Verbose = True
+```
+
+If we don't need to configure the tool or we want to use the defaults, there's no need for storing the returned tool in a variable.
+For example, if we wanted the information of the PV associated to our particle, we could just add the `TupleToolPrimaries` with no further configuration:
+
+```python
+dtt.addTupleTool('TupleToolPrimaries')
+```
+
+The way the `DecayTreeTuple.Decay` is written in in our [minimal DaVinci job](09-minimal-dv-job.html), 
+
+```python
+dtt.Decay = '[D*(2010)+ -> (D0 -> K- pi+) pi+]CC'
+```
+
+means that the configured `TupleTools` will only run on the head of the decay chain, that is, the D*.
+In order to select the particles for which we want the information stored, we need to mark them with a `^` symbol in the decay descriptor.
+For example, if we want to fill the information of the D0 and its children, we would modify the `dtt` to look like this:
+
+```python
+dtt.Decay = '[D*(2010)+ -> ^(D0 -> ^K- ^pi+) pi+]CC'
+```
+
+This will run all the configured `TupleTools` on the marked particles, with the caveat that some tools are only run on certain types of particles (eg, tracking tools on particles that have an associated track).
+This configuration is not optimal, since there may be tools which we only want to run on the D's and some only on the children. Enter `Branches`, which allow us to specify which tools get applied to which particle in the decay (in addition to the `TupleTools` configured at the top level).
+
+Branches are configured by means of a `dict` that relates the variable name and the decay descriptor needed to select it. This also allows to give more descriptive names to the particles, which will be used as the prefix of the tuple leaves (of the form `PARTICLENAME_VARNAME`)
```

Comment:
Maybe we should specifically spell out that the keys in the dict are the PARTICLENAMEs? (Might be worth calling them BRANCHNAMES, too)
This is probably very obvious, but it wouldn't hurt to write it down :)
As in 
> Branches let you define custom namespaces in your NTuple.
> They are configured by means of a `dict`.
> The keys of the `dict` let you define the name of each branch, while the corresponding values are decay descriptors that specify which particles you want to include in the branch.